plugins/inputs/statsd/statsd.go

```diff
@@ -129,6 +129,8 @@ type Statsd struct {
 
 	// A pool of byte slices to handle parsing
 	bufPool sync.Pool
+
+	MaxParserThreads int `toml:"max_parser_threads"`
```

Comment:
Let's not expose this option, since it is so tied to an implementation detail.  Let's just pick a good default number and always start this many.  How about 3?

Reply:
I'm not sure if I agree with that. It's definitely tied to an implementation detail, but I think there is value in allowing the level of concurrency to be configurable. For instance, if I choose to run telegraf on the same machine as my app, I may want to limit it's footprint. But if I set up a dedicated instance to act as a metrics aggregator, being able to spend more CPU to increase throughput (and reduce the number of servers I need to use) could be very useful.

Looking at the existing telegraf.conf, it seems as if there are a few places where this is configurable (but maybe we should rename it?)

https://github.com/influxdata/telegraf/blob/master/etc/telegraf.conf#L6124
https://github.com/influxdata/telegraf/blob/master/etc/telegraf.conf#L373
https://github.com/influxdata/telegraf/blob/master/etc/telegraf.conf#L4961

Maybe there's some way we could expose a level of concurrency as an option without tying it directly to the number of goroutines?

Reply:
I'm not confident enough yet that this is the right solution, at the very least we would need to do better performance testing before I'd be willing to add it.  I'd also like to discuss this with @reimda before we commit to any new options.